File: src/test/java/com/mihael/mhipster/domain/TestReportTestSamples.java
package com.mihael.mhipster.domain;

import java.util.Random;
import java.util.UUID;
import java.util.concurrent.atomic.AtomicInteger;
import java.util.concurrent.atomic.AtomicLong;

public class TestReportTestSamples {

    private static final Random random = new Random();
    private static final AtomicLong longCount = new AtomicLong(random.nextInt() + (2 * Integer.MAX_VALUE));
    private static final AtomicInteger intCount = new AtomicInteger(random.nextInt() + (2 * Short.MAX_VALUE));

    public static TestReport getTestReportSample1() {
        return new TestReport()
            .id(1L)
            .html("html1")
            .missedInstructions(1)
            .instructions(1)
            .missedBranches(1)
            .branches(1)
            .missedLines(1)
            .lines(1)
            .missedMethods(1)
            .methods(1)
            .missedClasses(1)
            .classes(1);
    }

    public static TestReport getTestReportSample2() {
        return new TestReport()
            .id(2L)
            .html("html2")
            .missedInstructions(2)
            .instructions(2)
            .missedBranches(2)
            .branches(2)
            .missedLines(2)
            .lines(2)
            .missedMethods(2)
            .methods(2)
            .missedClasses(2)
            .classes(2);
    }

    public static TestReport getTestReportRandomSampleGenerator() {
        return new TestReport()
            .id(longCount.incrementAndGet())
            .html(UUID.randomUUID().toString())
            .missedInstructions(intCount.incrementAndGet())
            .instructions(intCount.incrementAndGet())
            .missedBranches(intCount.incrementAndGet())
            .branches(intCount.incrementAndGet())
            .missedLines(intCount.incrementAndGet())
            .lines(intCount.incrementAndGet())
            .missedMethods(intCount.incrementAndGet())
            .methods(intCount.incrementAndGet())
            .missedClasses(intCount.incrementAndGet())
            .classes(intCount.incrementAndGet());
    }
}

File: src/main/webapp/app/entities/test-report/test-report.test-samples.ts
import { ITestReport, NewTestReport } from './test-report.model';

export const sampleWithRequiredData: ITestReport = {
  id: 2291,
};

export const sampleWithPartialData: ITestReport = {
  id: 3168,
  branches: 27246,
  missedMethods: 30989,
  missedClasses: 21708,
  classes: 14467,
};

export const sampleWithFullData: ITestReport = {
  id: 11099,
  html: 'through jittery',
  runtimeRetention: true,
  missedInstructions: 10410,
  instructions: 12698,
  missedBranches: 21748,
  branches: 25190,
  missedLines: 28493,
  lines: 23558,
  missedMethods: 9062,
  methods: 7436,
  missedClasses: 21940,
  classes: 27739,
};

export const sampleWithNewData: NewTestReport = {
  id: null,
};

Object.freeze(sampleWithNewData);
Object.freeze(sampleWithRequiredData);
Object.freeze(sampleWithPartialData);
Object.freeze(sampleWithFullData);
